### PR TITLE
all: Add/fix reference server health check and integration test.

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What
Add/fix reference server health check and integration test.

### Why

N/A

### Known limitations

Only KafkaListener is checked. We should add SQS checks.

### CODE REVIEW

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java [hunk 1]
```diff
@@ -87,4 +96,50 @@ public void destroy() {
...
+    return "kafka_listener";
+  }
+
+  @Override
+  public List<String> getTags() {
+    return List.of("all", "kafka");
+  }
+
+  @Override
+  public HealthCheckResult check() {
```
Comment (stfung77) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java:116]: should we test if the service is able to connect to kafka as well? ex: call consumer.listTopics() just to check the connection.

https://stackoverflow.com/questions/41016075/check-if-system-is-connected-to-kafka

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java [hunk 2]
```diff
@@ -87,4 +96,50 @@ public void destroy() {
...
+        .build();
+  }
+}
+
+@Data
+@Builder
+class KafkaHealthCheckResult implements HealthCheckResult {
+  transient String name;
+
+  List<String> statuses = List.of("green", "red");
```
Comment (marcelosalloum) [inline anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java:135]:
```suggestion
  List<String> statuses = List.of(GREEN.getName(), RED.getName());
```

File: integration-tests/src/main/kotlin/org/stellar/anchor/platform/ReferenceServerClient.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package org.stellar.anchor.platform
+
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.Request
+
+class ReferenceServerClient(private val endpoint: String) : SepClient() {
+  fun health(checks: List<String>): java.util.HashMap<*, *> {
```
Comment (marcelosalloum) [inline integration-tests/src/main/kotlin/org/stellar/anchor/platform/ReferenceServerClient.kt:8]: 💡 You could default the health check test to `"all"`:

```suggestion
  fun health(checks: List<String> = listOf("all")): java.util.HashMap<*, *> {
```

**Review by marcelosalloum — APPROVED** [review]

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java (hunk 1 above)
Comment (lijamie98) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java:116]: This is great suggestion. The check is added.

File: anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java (hunk 2 above)
Comment (lijamie98) [reply anchor-reference-server/src/main/java/org/stellar/anchor/reference/event/KafkaListener.java:135]: Thanks for catching this. Corrected.

File: integration-tests/src/main/kotlin/org/stellar/anchor/platform/ReferenceServerClient.kt (hunk 3 above)
Comment (lijamie98) [reply integration-tests/src/main/kotlin/org/stellar/anchor/platform/ReferenceServerClient.kt:8]: Nice. Changed.

**Pull request merged**